internal/controller/object/object.go

```diff
@@ -65,6 +65,13 @@ const (
 	errGetLastApplied          = "cannot get last applied"
 	errUnmarshalTemplate       = "cannot unmarshal template"
 	errFailedToMarshalExisting = "cannot marshal existing resource"
+
+	errGetReferencedResource       = "cannot get referenced resource"
+	errPatchFromReferencedResource = "cannot patch from referenced resource"
+	errResolveResourceReferences   = "cannot resolve resource references"
+
+	errFailedToAddFinalizer    = "Failed to add finalizer to referenced resource"
+	errFailedToRemoveFinalizer = "Failed to remove finalizer to referenced resource"
```

Comment:
error messages should start with lowercase.

nitpick: lets use `cannot` instead of `failed to` just for consistency

```suggestion
	errFailedToAddFinalizer    = "cannot add finalizer to referenced resource"
	errFailedToRemoveFinalizer = "cannot remove finalizer to referenced resource"
```